src/renderer/components/+config-autoscalers/hpa-v1-metric-parser.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import type { MetricCurrentTarget, V2Beta1ContainerResourceMetricSource, V2Beta1ContainerResourceMetricStatus, V2Beta1ExternalMetricSource, V2Beta1ExternalMetricStatus, V2Beta1ObjectMetricSource, V2Beta1ObjectMetricStatus, V2Beta1PodsMetricSource, V2Beta1PodsMetricStatus, V2Beta1ResourceMetricSource, V2Beta1ResourceMetricStatus } from "../../../common/k8s-api/endpoints";
+
+export class HorizontalPodAutoscalerV1MetricParser {
```

Comment:
Why is this a class and not several functions?

Reply:
For keeping same names for class methods within `HorizontalPodAutoscalerV1MetricParser` and `HorizontalPodAutoscalerV2MetricParser`.

Keeping them as functions will require to name each of them like

```
function getV2MetricsResource({ current, target }: { current: V2ResourceMetricStatus | undefined; target: V2ResourceMetricSource }): MetricCurrentTarget {
```

Reply:
ok sounds good